src/function/aggregate/distributive/mode.cpp

```diff
@@ -51,14 +51,15 @@ struct ModeFunction {
 			mask.SetInvalid(idx);
 			return;
 		}
-		T h_freq;
-		size_t freq = 0;
-
-		for (auto &val : *state->frequency_map) {
-			if (val.second > freq) {
-				h_freq = val.first;
-				freq = val.second;
+		auto val = state->frequency_map->begin();
+		T h_freq = val->first;
+		size_t freq = val->second;
```

Comment:
Why did you initialise `freq` to `val->second` now and not to `0` as before? Otherwise looks good!

Reply:
I'm basically initializing them to whatever is the first value in the ht and then comparing with the others, I guess I should've incremented the iterator before going to the while loop to save on one comparison as well, but I think the initialization itself is correct, I think it is correct, no?